libpromises/cf3lex.l

```diff
@@ -133,6 +138,57 @@ promise_type   [a-zA-Z_]+:
                           P.line_pos = 0;
                       }
 
+{macro_if}            {
+                        const char* version_text = yytext+18;
+                        ParserDebug("\tL:macro @if %d:version=%s\n", P.line_pos, version_text);
+                        {
+                          int request_major = 0;
+                          int request_minor = 0;
+                          int major = 0;
+                          int minor = 0;
+                          int patch = 0;
+                          if (3 == sscanf(Version(), "%d.%d.%d", &major, &minor, &patch))
+                          {
+                            if (2 == sscanf(version_text, "%d.%d", &request_major, &request_minor))
+                            {
+                              if (request_major > major || request_minor > minor)
```

Comment:
This expression will make ``@if after_version(2.8)`` ignore the block on 3.7. You'll need:

<pre>
if (request_major > major || (request_major == major && request_minor > minor))
</pre>